scripts/coccinelle/identifier_length.cocci

```diff
@@ -3,6 +3,20 @@
 
 virtual report
 
+@initialize:python
+  depends on report
+@
+@@
+identifiers = {}
+
+def check_identifier(id, pos):
+    if id in identifiers:
+       msg="WARNING: Violation to rule 5.1 or 5.2 (First 31 chars shall be distinct) %s" % (id)
+       coccilib.report.print_report(pos[0], msg)
+       coccilib.report.print_report(identifiers[id][0], msg)
```

Comment:
Will this line print the msg content but with first found colliding identifier?

Reply:
will print the first occurrence and the one just found, the reason to print it this way is that the output of this script is consumed later by another script that parses this pattern.

Reply:
ok